stan/math/rev/fun/inc_beta.hpp

```diff
@@ -19,17 +18,84 @@ class inc_beta_vvv_vari : public op_vvv_vari {
   inc_beta_vvv_vari(vari* avi, vari* bvi, vari* cvi)
       : op_vvv_vari(inc_beta(avi->val_, bvi->val_, cvi->val_), avi, bvi, cvi) {}
   void chain() {
-    double d_a;
-    double d_b;
-    const double beta_ab = beta(avi_->val_, bvi_->val_);
-    grad_reg_inc_beta(d_a, d_b, avi_->val_, bvi_->val_, cvi_->val_,
```

Comment:
Is the intention to remove `grad_reg_inc_beta()`?

Reply:
I think originally I just wanted to split the rev-mode implementation into three specializations, because it was propagating gradients it didn't need too ... and then I found out that grad_reg_inc_beta seems to have been superseded and I rewrote this code to avoid it.

But honestly I no longer think whatever benefit was to be had from these changes is worthwhile.

I appreciate you taking the time to review, sorry for the noise.